benches/benches/bevy_transform/transform_hierarchy.rs

```diff
@@ -0,0 +1,655 @@
+    const ASYNC_PLUGIN_INIT: bool = false;
+    if ASYNC_PLUGIN_INIT {
+        while !app.ready() {
+            #[cfg(not(target_arch = "wasm32"))]
+            bevy_tasks::tick_global_task_pools_on_main_thread();
+        }
+    }
+    assert!(app.ready());
+
+    app.finish();
+    app.cleanup();
+
+    // Run setup (what would normally happen in the Startup schedule)
+    setup(&mut app.world, cfg);
+
+    app
+}
+
+fn bench_group<M: criterion::measurement::Measurement>(mut group: BenchmarkGroup<M>, cfg: &Cfg) {
+    group.warm_up_time(std::time::Duration::from_secs(2));
+    group.measurement_time(std::time::Duration::from_secs(10));
+
+    // This benchmark tries to measure the cost of the initial transform propagation,
+    // i.e. the first time transform propagation runs after we just added all our entities.
+    group.bench_function("transform_init", |b| {
+        // Building the World (in setup) takes a lot of time, so we shouldn't do that on every
+        // iteration.
+        // Unfortunately, we can't re-use an App directly in iter() the World would no longer be
+        // in its pristine, just initialized state from the second iteration onwards.
+        // Furthermore, it's not possible to clone a pristine World since World doesn't implement
+        // Clone.
+        // As an alternative, we reuse the same App and reset it to a pseudo-pristine state by
+        // simply marking all Parent, Children and Transform components as changed.
+        // This should look like a pristine state to the propagation systems.
+
+        let mut app = build_app(cfg, false);
+
+        app.add_schedule(ResetSchedule, reset_schedule());
+
+        // UnsafeWorldCell is necessary here since we'd otherwise pass a &mut World/&mut App to both
+        // iter_batched closures.
+        let cell = app.world.as_unsafe_world_cell();
+
+        b.iter_batched(
+            || {
+                // reset_schedule.run(unsafe { cell.world_mut() });
+                unsafe { cell.world_mut() }.run_schedule(ResetSchedule);
+
+                cell
+            },
+            |cell| {
+                unsafe { cell.world_mut() }.run_schedule(bevy_app::Main);
+            },
+            BatchSize::PerIteration,    
+        );
+    });
+
+    // Measures hierarchy propagation systems when there are no changes
+    // during the Update schedule.
+    group.bench_function("transform_update_noop", |b| {
+        let mut app = build_app(cfg, false);
+
+        // Run Main schedule once to ensure initial updates are done
+        app.update();
+
+        b.iter(move || { app.update(); });
+    });
+
+    // Measures hierarchy propagation systems when some transforms are updated.
+    group.bench_function("transform_update", |b| {
+        let mut app = build_app(cfg, true);
+
+        // Run Main schedule once to ensure initial updates are done
+        app.update();
+
+        b.iter(move || { app.update(); });
```

Comment:
This also measures the `update` system runtime right?

Reply:
No, the names are a little misleading here: `App::update` is simply what usually gets called by the `app.runner` (the closure set by the ScheduleRunnerPlugin/winit), and in this case it simply runs the main schedule since we the benchmark app doesn't have any subapps.
And the initial updates I'm referring to in the first comment are both whatever happens in the `Startup` schedule and the first time the propagations are run in `PostUpdate`. I didn't specify that in the comment as I'm trying to keep the benchmark implementation-agnostic, perhaps the propagation systems will be run in a different Schedule in the future.

Reply:
Sorry, just realized I misread your comment: Yes, the update system runtime is included here.